core/engine/src/object/builtins/jstypedarray.rs

```diff
@@ -273,6 +358,104 @@ impl JsTypedArray {
         )
     }
 
+    /// Calls `TypedArray.prototype.findIndex()`.
```

Comment:
nit: can there be some examples added to these docs.

Since `JsTypedArray` et al. are meant to be an externally facing API. Some more thorough examples in the documentation would be nice. Something a bit more like the examples provided in [`JsPromise`](https://boajs.dev/boa/doc/boa_engine/object/builtins/struct.JsPromise.html) or [`JsMap`](https://boajs.dev/boa/doc/boa_engine/object/builtins/struct.JsMap.html).

Reply:
Do you think it would be more meaningful to enhance the [builtins/jstypedarray.rs](https://github.com/AngeloChecked/boa/blob/JsTypedArray_methods/core/engine/src/object/builtins/jstypedarray.rs) module with standard Rust documentation, or should we focus solely on improving the code in the [examples](https://github.com/AngeloChecked/boa/blob/JsTypedArray_methods/examples/src/bin/jsarraybuffer.rs) folder? I drew a lot of inspiration from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/findIndex) for those examples, because originally I hadn't thought about it; I used them as acceptance tests to ensure the implementation was correct. However, I'm open to reconsidering them as documentation if that would be more beneficial.

Reply:
The convention is to have complex examples about general usage on `examples`, and simple examples as documentation comments.

Reply:
The rule of thumb is to have examples that demonstrate HOW to use the APIs as doc comments, and have examples that demonstrate WHEN to use the APIs as code examples.

Reply:
The module level `rustdoc` documentation was more what I meant in this instance. The examples don't have to be a completely different from those in `examples`. Mostly since the Rust documentation will be auto generated into the docs, which is super handy and may be the first place someone consuming the API would go for information. For example, here's the [`JsPromise` docs](https://boajs.dev/boa/doc/boa_engine/object/builtins/struct.JsPromise.html) for reference

Reply:
ok, [here](https://github.com/boa-dev/boa/pull/3481/commits/703411b32d8343bcadec43594a9451b0ed1864df#diff-1173b2fbe95c226df7883fdf2e2b9c0e40d2270b6ee2f58053b0f3ef38bce345R361) i created the first rust doc test

the preview looks like:
![image](https://github.com/boa-dev/boa/assets/30749948/e6711739-42de-4e64-bf36-576e30adee7e)

I intend to keep drawing considerable inspiration from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/findIndex) for comments. What are your thoughts? Should I proceed with implementing Rust doc tests throughout the module? Do you have any specific considerations or tips to share?

Reply:
Looks really nice! I think it's a good call to adapt the examples from MDN, since it removes the effort of having to come up with descriptive usages.

As a small suggestion, I would remove the "The FindIndex method of Array instances" prefix, since it is mostly idiomatic to start a method's documentation with a verb; in this case, "Returns the index of...".

Reply:
I've added doc tests to all the functions I've touched. Give me feedback!